agent/bench-scripts/pbench-linpack

```diff
@@ -187,8 +187,17 @@ function record_iteration {
 	echo ${iteration} | pbench-add-metalog-option ${mdlog} iterations/${iteration} iteration_name
 }
 
+mkdir ${benchmark_run_dir}
+if [[ ${?} -ne 0 ]]; then
```

Comment:
There is a great redundancy, perhaps you can refactor this into a function...

Reply:
> (also not sure what is wrong with the `-p`, but I understand you want to be safe)

Right, `mkdir -p` will create all paths from `/` down since most of the time we are running as `root` and we use the resolved paths.

Reply:
> There is a great redundancy, perhaps you can refactor this into a function...

That is a possibility but the error message is unique, so there does not appear to be much gain unless we make the error message generic.  And it seems a trade-off worth making for the context provided to the user.